test/wallet-account.test.js

```diff
@@ -185,3 +202,69 @@ it('requests transaction signing automatically when there is no local key', asyn
     expect(transactions[0].actions[0].transfer.deposit.toString()).toEqual('1');
     expect(Buffer.from(transactions[0].publicKey.data)).toEqual(Buffer.from(keyPair.publicKey.data));
 });
+
+it('requests transaction signing automatically when function call has attached deposit', async () => {
+    let localKeyPair = nearApi.KeyPair.fromRandom('ed25519');
+    let walletKeyPair = nearApi.KeyPair.fromRandom('ed25519');
+    setupWalletConnectionForSigning({
+        allKeys: [ walletKeyPair.publicKey.toString() ],
+        accountAccessKeys: [{
+            access_key: {
+                nonce: 1,
+                permission: {
+                    FunctionCall: {
+                        allowance: '1000000000',
+                        receiver_id: 'receiver.near',
+                        method_names: []
+                    }
+                }
+            },
+            public_key: localKeyPair.publicKey.toString()
+        }, {
+            access_key: {
+                nonce: 1,
+                permission: 'FullAccess'
```

Comment:
Is it an important part of what you're testing that this test sets this `FullAccess` key and the next test does not?